src/hiccup/form.clj

```diff
@@ -74,6 +74,19 @@
              :value value
              :checked checked?}]))
 
+;; According to the HTML spec, the value inside the boolean `selected` attribute
+;; can only be "" or "selected" (case insensitive).  Hence, the coercion to
+;; boolean is required (otherwise, the attribute value will be `x` itself).
+;; We also need the `(not (sequential? selected))` requirement in the
+;; second `and` because if `selected` is a vector where `x` is not an element,
+;; the first `and` will fail, and the second will try to run something
+;; like `([a b] c)`, and can easily lead to java.lang.IndexOutOfBoundsException.
+(defn- selected? [x selected]
+  (or
+    (and (sequential? selected) (boolean ((set selected) x)))
+    (and (not (sequential? selected)) (ifn? selected) (selected x))
+    (= x selected)))
```

Comment:
Given the purpose of this expression, perhaps a better option is to use `cond`:

```clojure
(defn- selected? [x selected]
  (cond
    (sequential? selected) (boolean ((set selected) x)))
    (ifn? selected) (selected x))
    :else (= x selected)))

Reply:
Makes sense.  Tests pass as well.